homeassistant/components/tplink/entity.py

```diff
@@ -287,101 +284,112 @@ def _description_for_feature[_D: EntityDescription](
                 translation_key = None
                 entity_name = child_alias  # if None will use device name
 
-            desc = replace(
+            return replace(
                 desc,
                 translation_key=translation_key,
                 name=entity_name,  # if undefined will use translation key
-                entity_category=category,
+                entity_category=cls._category_for_feature(feature),
                 # enabled_default can be overridden to False in the description
-                entity_registry_enabled_default=(not is_debug_feature)
+                entity_registry_enabled_default=feature.category
+                is not Feature.Category.Debug
                 and desc.entity_registry_enabled_default,
             )
-        else:
-            desc = desc_cls(
-                key=feature.id,
-                name=feature.name,
-                icon=feature.icon,
-                entity_category=category,
-                entity_registry_enabled_default=not is_debug_feature,
+
+        _LOGGER.info(
+            "Device feature: %s (%s) needs an entity description defined in HA",
+            feature.name,
+            feature.id,
+        )
+        return None
+
+    @classmethod
+    def _entities_for_device[
+        _E: CoordinatedTPLinkFeatureEntity,
+        _D: TPLinkFeatureEntityDescription,
+    ](
+        cls,
+        device: Device,
+        coordinator: TPLinkDataUpdateCoordinator,
+        *,
+        feature_type: Feature.Type,
+        entity_class: type[_E],
+        descriptions: Mapping[str, _D],
+        parent: Device | None = None,
+    ) -> list[_E]:
+        """Return a list of entities to add.
+
+        This filters out unwanted features to avoid creating unnecessary entities
+        for device features that are implemented by specialized platforms like light.
+        """
+        entities: list[_E] = [
+            entity_class(
+                device,
+                coordinator,
+                feature=feat,
+                description=desc,
+                parent=parent,
             )
-            _LOGGER.warning(
-                "Device feature: %s (%s) needs an entity description defined, "
-                "it has been added to hass but may not be presented properly",
-                feature.name,
-                feature.id,
+            for feat in device.features.values()
+            if feat.type == feature_type
```

Comment:
If this is an enum on both sides we can use is

Reply:
Yes it is